endpointmanager/pkg/historypruning/historypruning.go

```diff
@@ -0,0 +1,92 @@
+package historypruning
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+
+	"github.com/lib/pq"
+	"github.com/onc-healthit/lantern-back-end/endpointmanager/pkg/capabilityparser"
+	"github.com/onc-healthit/lantern-back-end/endpointmanager/pkg/endpointmanager/postgresql"
+	"github.com/onc-healthit/lantern-back-end/endpointmanager/pkg/helpers"
+)
+
+// PruneInfoHistory checks info table and prunes any repetitive entries
+func PruneInfoHistory(ctx context.Context, store *postgresql.Store, queryInterval bool) {
+
+	var rows *sql.Rows
+	var err error
+
+	rows, err = store.PruningGetInfoHistory(ctx, queryInterval)
+	helpers.FailOnError("", err)
+
+	if !rows.Next() {
+		return
+	}
+
+	_, fhirURL1, _, capStat1, tlsVersion1, mimeTypes1, smartResponse1 := getRowInfo(rows)
+
+	for rows.Next() {
+
+		operation2, fhirURL2, entryDate2, capStat2, tlsVersion2, mimeTypes2, smartResponse2 := getRowInfo(rows)
+
+		// If capstat is not null check if current entry that was passed in has capstat equal to capstat of old entry being checked from history table, otherwise check they are both null
+		var capStatEqual bool
+		var smartResponseEqual bool
+
+		tlsVersionEqual := (tlsVersion1 == tlsVersion2)
+		mimeTypesEqual := helpers.StringArraysEqual(mimeTypes1, mimeTypes2)
+
+		if capStat1 != nil {
+			capStatEqual = capStat1.EqualIgnore(capStat2)
+		} else {
+			capStatEqual = (capStat2 == nil)
+		}
+
+		if smartResponse1 != nil {
+			smartResponseEqual = smartResponse1.EqualIgnore(smartResponse2)
+		} else {
+			smartResponseEqual = (smartResponse2 == nil)
+		}
+
+		equal := capStatEqual && tlsVersionEqual && mimeTypesEqual && smartResponseEqual
+
+		if equal && operation2 != "I" {
```

Comment:
I think this should check for operation being == "U" that way we don't accidentally remove deletion operations